src/Microsoft.AspNet.Authentication.OAuth/OAuthHandler.cs

```diff
@@ -155,16 +155,17 @@ public OAuthHandler(HttpClient backchannel)
 
         protected virtual async Task<AuthenticationTicket> CreateTicketAsync(ClaimsIdentity identity, AuthenticationProperties properties, OAuthTokenResponse tokens)
         {
-            var context = new OAuthCreatingTicketContext(new ClaimsPrincipal(identity), properties, Context, Options, Backchannel, tokens);
+            var ticket = new AuthenticationTicket(new ClaimsPrincipal(identity), properties, Options.AuthenticationScheme);
+            var context = new OAuthCreatingTicketContext(ticket, Context, Options, Backchannel, tokens);
 
             await Options.Events.CreatingTicket(context);
 
-            if (context.Principal?.Identity == null)
+            if (context?.Ticket?.Principal?.Identity == null)
```

Comment:
nit: `context` cannot be null. That said, this check is probably not necessary now.